Opentype/Fileformat/Name.hs

```diff
@@ -97,8 +97,8 @@ readNameTable bs = do
     Right (offset, len2, NameRecord pf enc lang nID)
   records2 <- for records $
     \(offset, len2, r) ->
-      if storage+offset+len2 > fromIntegral (Strict.length bs)
-        then Left "string storage bounds exceeded"
+      if fromIntegral (storage+offset+len2) > Strict.length bs
+        then Left $ "overflow error: name record in storage at (" <> show storage <> ") at offset (" <> show offset <> ") with length (" <> show len2 <> ") exceeds input length (" <> show (Strict.length bs) <> ")"
```

Comment:
1. By converting the offsets to `Int` instead of converting the `ByteString` to `Word16`, we can deal with inputs longer than 2^16.

2. Possibly we should be converting the `storage`, `offset`, & `len2` variables to `Int` separately, rather than converting the sum. Similarly, we should maybe convert `offset` and `storage` before adding them in the argument to `Strict.drop`? I’m not sure of the precise semantics for these fields or whether their sums can overflow, tho.

3. I added a bunch of error details to debug this and I figured that might still be helpful context.